src/App.vue

```diff
@@ -104,36 +134,51 @@ export default {
       'setNextARDate', 'setTasks', 'setFilings', 'setRegisteredAddress', 'setRecordsAddress', 'setDirectors',
       'setLastAnnualReportDate', 'setConfigObject']),
 
-    fetchData () {
+    /** Stores today's date and sets timeout to keep it updated. */
+    updateCurrentDate (): void {
+      // clear previous timeout, if any
+      clearTimeout(this.currentDateTimerId)
+
+      // set current date
+      const now = new Date()
+      const date = this.dateToUsableString(now)
+      this.setCurrentDate(date)
+
+      // set timeout to run this again at midnight
+      const hoursToMidnight = 23 - now.getHours()
+      const minutesToMidnight = 59 - now.getMinutes()
+      const secondsToMidnight = 59 - now.getSeconds()
+      const timeout = ((((hoursToMidnight * 60) + minutesToMidnight) * 60) + secondsToMidnight) * 1000
+      this.currentDateTimerId = setTimeout(this.updateCurrentDate, timeout)
```

Comment:
Is all of this in the event a client leaves the browser open? Just curious as to the purpose of implementation!

Reply:
* Edit, i see this was all existing prior.

Reply:
Yes, just moved this up. Also added a safety check to ensure we don't get multiple timers.